cli/src/commands/extensions/mod.rs

```diff
@@ -0,0 +1,113 @@
+pub mod extension;
+
+use std::{collections::HashSet, convert::TryFrom, path::PathBuf};
+
+use crate::commands::{CommandResult, CommandValue, ExitCode};
+pub use extension::*;
+use log::{error, info};
+
+use anyhow::Result;
+use clap::{arg, ArgMatches, Command, ValueHint};
+
+pub fn command<'a>() -> Command<'a> {
+    Command::new("extension")
+        .about("Run extensions")
+        .subcommand(
+            Command::new("add")
+                .about("Install extension")
+                .arg(arg!([PATH]).required(true).value_hint(ValueHint::FilePath)),
+        )
+        .subcommand(
+            Command::new("remove")
+                .about("Uninstall extension")
+                .arg(arg!([NAME]).required(true)),
+        )
+        .subcommand(Command::new("list").about("List installed extensions"))
+}
+
+/// Generate the subcommands for each extension.
+/// TODO add tests.
+pub fn extensions_subcommands(command: Command<'_>) -> Command<'_> {
+    let extensions = match list_extensions() {
+        Ok(extensions) => extensions,
+        Err(e) => {
+            error!("Couldn't list extensions: {}", e);
+            return command;
+        }
+    };
+
+    let names = command
+        .get_subcommands()
+        .map(|n| n.get_name().to_string())
+        .collect::<HashSet<_>>();
+
+    extensions
+        .into_iter()
+        .filter(|ext| !names.contains(ext.name()))
```

Comment:
It might be nice to have a warning here for any extensions that get filtered out.